beer-garden/beer-garden
# #1182 - Generate requirements.txt from setup.py

Closes #1182 

This PR slightly changes the way we manage the project dependencies.  It does the following:

* Updated the version pinning in the `setup.py` to match what's currently in the `requirements.in`.
* Moved the test and dev dependencies from the `requirements.in` to a `requirements-dev.in`
* Removed `requirements.in`
* Added a `deps-compile` command to the Makefile that generates the `requirements.txt` and a new `requirements-dev.txt` from the `setup.py` and `requirements-dev.in` respectively.
* Updated the `deps` command in the Makefile so that it now installs both the `requirements.txt` and `requirements-dev.txt` dependencies.

The net effect of all of this is that we now only need to manage the dependencies for running beer garden in the setup.py.  Any dependencies specific to development and testing are managed separately int he `requirements-dev.in`.

## Test Instructions
* Remove the `requirements.txt` and `requirements-dev.txt`
* From the `/src/app` directory, run `make deps-compile`
* This should recreate the `requirements.txt` and `requirements-dev.txt` files for you.
* Running `make deps` should do a pip install using those two files for you.
